### DISCUSSION TITLE
Naming conventions: `Vec3::as_dvec3` vs `Quat::as_f64`

### DISCUSSION THREAD

**Discussion by LPGhatguy** [bitshifter/glam-rs#446] [General]
Hello! Today I'm starting a port of our game from using f32 coordinates to using f64 coordinates for everything in world space. We're switching a lot of `Vec3` values to `DVec3`, and `Quat` to `DQuat`.

As part of that work, I noticed that conversions in the vector types and the quaternion types have different naming conventions! I thought that was incongruous, so I wanted to open this thread to ask about that.

I like the naming convention in the `Quat` types. I think `as_<numtype>` is more clear than `as_<vectortype>` in code that's converting both vectors and quaternions.

One consideration is that there are many more numeric types supported for vectors, as well as the 16-byte aligned SIMD variants with the `A` suffix. Normalizing the naming scheme to match `Quat` would make those other conversions stand out.

What are you thoughts on this? Do you think it would be valuable to change either of the naming conventions, i.e. adding `Quat::as_dquat` or adding `Vec3::as_f64`?

**Comment by bitshifter** [comment]
I think the history of these methods may be lost as they started on a dev branch that has been merged and deleted. From memory though, they started out as `as_f64`, etc and were later renamed to `as_dvec3` and so on. It looks like I missed the `Quat` methods as part of that rename.

It was a while ago that I was thinking about this, but I think I ultimately felt that `as_dvec3` was a lot more explicit about what was happening than say `as_f64`, the later is a scalar value so probably people would assume it would create a vector of `f64` but maybe it's not completely obvious. Then there's also the extra `align(16)` types to deal with and it seemed more convenient to add something that converted directly to those, although `as_vec3a` might be a bit misleading there as there's more going on than simply casting the properties.

I think it would make sense to align the `Quat` methods with everything else and deprecate the `as_f32` and `as_f64` on quat types.
